std/zip.d

```diff
@@ -862,12 +1070,55 @@ public:
                 import std.zlib : uncompress;
                 de._expandedData = cast(ubyte[]) uncompress(cast(void[]) de.compressedData, de.expandedSize, -15);
                 return de.expandedData;
-
-            default:
-                throw new ZipException("unsupported compression method");
```

Comment:
I think this will now throw an `Error` instead of `Exception`, so this change is wrong.

Also not sure about the other similar change for similar reasons.

Reply:
Good point. I'll add a unittest here. When I remember right (I started preparing this about a month ago), I did not manage to cover the other place (but I'll try again) and therefore changed to the final switch. Here I just thought, it's the same, but that is not true.